src/compiler/crystal/syntax/to_s.cr

```diff
@@ -968,9 +968,9 @@ module Crystal
       @str << "/"
       case exp = node.value
       when StringLiteral
-        @str << exp.value.gsub('/', "\\/")
+        Regex.append_source exp.value, @str
       when StringInterpolation
-        visit_interpolation exp, &.gsub('/', "\\/")
+        visit_interpolation(exp) { |s| escape_regex s }
```

Comment:
Why not use `Regex.append_source` directly? `visit_interpolation` simply appends to `@str`. You can do that in the block as well and don't need the string builder. Just make sure the block returns `nil` or empty string.

Reply:
You are trick star!! Yeah!